### PR TITLE
#385 Partial Bugfix

### PR DESCRIPTION
**Pull request by PauloAmora**
This PR resolves issue #385.

In` SimpleOptimizer`: Adding the order by column explicitely into the column map, to allow materialization and sorting.

In `OrderByExecutor`: Use only the columns in select for output_schema.

As requested, only the necessary commit.

### CODE REVIEW

**Comment by PauloAmora** [comment]
I just figured that the modification done on `SimpleOptimizer` must be applied in 2 places. There is an `order and limit` condition besides the order condition. I tested order and limit with the modification and it is working.

File: src/executor/order_by_executor.cpp [hunk 1]
```diff
@@ -57,7 +57,8 @@ bool OrderByExecutor::DExecute() {
   PL_ASSERT(input_tiles_.size() > 0);
 
   // Returned tiles must be newly created physical tiles,
-  // which have the same physical schema as input tiles.
+  // which not always have the same physical schema as input tiles.
```
Comment (jarulraj) [inline src/executor/order_by_executor.cpp]: We can rephrase this as: "NOTE: the schema of these tiles might not match the input tiles when some of the order by columns are not be part of the output schema"

File: src/optimizer/simple_optimizer.cpp [hunk 2]
```diff
@@ -288,11 +297,13 @@ std::shared_ptr<planner::AbstractPlan> SimpleOptimizer::BuildPelotonPlanTree(
               ((expression::TupleValueExpression*)select_stmt->order->expr)
                   ->GetColumnName());
           for (size_t column_ctr = 0;
-               column_ctr < select_stmt->select_list->size(); column_ctr++) {
-            std::string col_name((
-                (expression::TupleValueExpression*)select_stmt->select_list->at(
-                    column_ctr))->GetColumnName());
-            if (col_name == sort_col_name) key.push_back(column_ctr);
+               column_ctr < target_table->GetSchema()->GetColumns().size(); column_ctr++) {
```
Comment (jarulraj) [inline src/optimizer/simple_optimizer.cpp]: The original code looks ugly with long lines. Can you break it up?
Like `auto schema_size = target_table->GetSchema()->GetColumns().size()` and `column_ctr < schema_size` etc.
`target_table->GetSchema()->GetColumns().at(column_ctr)`.

File: src/optimizer/simple_optimizer.cpp [hunk 3]
```diff
@@ -288,11 +297,13 @@ std::shared_ptr<planner::AbstractPlan> SimpleOptimizer::BuildPelotonPlanTree(
...
+            std::string col_name(target_table->GetSchema()->GetColumns().at(column_ctr).GetName());
+            if (col_name == sort_col_name) {
+                //The column_ctr is not reliable anymore given that we are looking to the whole schema.
+                //Since the columns were added in the column_ids, it is safe to retrieve only those indexes.
+                key.push_back(find(column_ids.begin(), column_ids.end(), column_ctr) - column_ids.begin());
```
Comment (jarulraj) [inline src/optimizer/simple_optimizer.cpp]: Store the iterator `find(column_ids.begin(), column_ids.end(), column_ctr)` in a variable `auto iter`.
Then use `std::distance(column_ids.begin(), iter)` instead of `iter - column_ids.begin()` to find the `offset`
Then push the `offset` into the `vector`.

Comment (jarulraj) [inline src/optimizer/simple_optimizer.cpp]: This PR would be perfect if you can also add a test case using `SQLTestsUtil` -- a query where a couple of `ORDER BY` columns are not part of the `SELECT` list.

**Comment by PauloAmora** [comment]
Ok, I'll work on that.

File: src/optimizer/simple_optimizer.cpp (hunk 3 above)
Comment (PauloAmora) [inline src/optimizer/simple_optimizer.cpp]: Where can I write those tests? I saw a `OptmizerSqlTest`, but I'm not sure if that is the correct class. I tried to find documentation on how to add test cases but couldn't find anything.

Comment (jarulraj) [inline src/optimizer/simple_optimizer.cpp]: Hey @PauloAmora -- here's an example: https://github.com/AngLi-Leon/peloton/blob/6c11ff7489d088d31b47cf99a12a6ce0ebb52626/test/sql/drop_sql_test.cpp. We can modify it based on our requirements.

**Comment by PauloAmora** [comment]
Something went wrong in my end, I'll add a new pull request
